frontend/apps/filemanager/filemanager.lua

```diff
@@ -157,6 +157,30 @@ function FileManager:init()
                         UIManager:close(self.file_dialog)
                     end,
                 },
+                {
+                    text = _("Purge .sdr"),
+                    enabled = lfs.attributes(util.realpath(file), "mode") == "file" and util.realpath(file):match("%.") and
+                            lfs.attributes(util.realpath(file):match("(.*)%.")..".sdr", "mode") == "directory",
```

Comment:
Thinking again, in the context of a document ("DocSettings"), the existing code should be fine.
In my context of FileManager, where i should find only directories and documents (with extensions), my tests could be simplified to : 
if lfs.attributes(full_path, "mode") == "file" and lfs.attributes(DocSettings:getSidecarDir(full_path), "mode") == "directory" then
But indeed, hasSidecarDir() is smaller. I'll do that tomorrow.

For getSidecar() returning nil, wouldnt we need to check for side effects in all the existing code, they probably expect a string.
Or just define how should be named the sidecar dir for a file with no extension we will never meet (just add .sdr ?)